pipeline/transform.py

```diff
@@ -0,0 +1,184 @@
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor, execute_values
+
+
+load_dotenv()
+
+SCRIPT_NAME = (os.path.basename(__file__)).split(".")[0]
+LOGGING_LEVEL = logging.DEBUG
+logger = cg.setup_logging(SCRIPT_NAME, LOGGING_LEVEL)
+
+
+class S3bucket:
+    """
+    downloads files in an S3 bucket
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize class variables.
+        """
+        self.logger = logger
+        self.bucket = "c12-energy-tracker"
+        self.public_key = os.environ.get("ACCESS_KEY")
+        self.secret_key = os.environ.get("SECRET_KEY")
+
+    def get_client(self) -> boto3.client:
+        """Connects to the S3 instance and returns it"""
+        s3 = boto3.client('s3',
+                          aws_access_key_id=self.public_key,
+                          aws_secret_access_key=self.secret_key)
+        return s3
+
+    def get_files_from_bucket(self):
+        """
+        Downloads All files in the s3 Bucket
+        """
+        client = self.get_client()
+        data = client.list_objects(Bucket=self.bucket)
+        for file in data['Contents']:
+
+            client.download_file(self.bucket, file['Key'], f'''{
+                                 file['Key']} {file['LastModified']}''')
+            self.logger.info(f"""Downloaded file: '{file['Key']} {
+                             file['LastModified']}'""")
+
+
+class Transform:
+    """
+   turns all files into a pd.Dataframe with the correct format
+
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize class variables.
+        """
+        self.logger = logger
+
+    def get_data(self) -> dict[tuple]:
+        files = [file for file in glob.glob(
+            f"*.feather*") if os.path.isfile(file)]
+        data = {}
+        for file in files:
+            df = pd.read_feather(file)
+            if "generation" in file:
+                formatted_data = self.generation_transform(df)
+                data['generation'] = formatted_data
+                self.logger.info(
+                    f"""Successfully transformed generation data""")
+            if "demand" in file:
+                formatted_data = self.demand_transform(df)
+                data['demand'] = formatted_data
+                self.logger.info(f"""Successfully transformed demand data""")
+            if "cost" in file:
+                formatted_data = self.cost_transform(df)
+                data['cost'] = formatted_data
+                self.logger.info(f"""Successfully transformed cost data""")
+            if "carbon" in file:
+                formatted_data = self.carbon_transform(df)
+                data['carbon'] = formatted_data
+                self.logger.info(f"""Successfully transformed carbon data""")
+
+        self.delete_read_files(files)
+        return data
+
+    def generation_transform(self, df: pd.DataFrame) -> tuple:
+        df['publish_date'] = df['publishTime'].apply(lambda x: x.split('T')[0])
```

Comment:
I cannot look at this and know what it does so we should make tests next week so that people like me don't break things.